src/oracles/CIXPythAdapter.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: BUSL-1.1
+// This code is made available under the terms and conditions of the Business Source License 1.1 (BUSL-1.1).
+// The act of publishing this code is driven by the aim to promote transparency and facilitate its utilization for educational purposes.
+
+pragma solidity 0.8.18;
+
+import { ABDKMath64x64 } from "@hmx/libraries/ABDKMath64x64.sol";
+import { HMXLib } from "@hmx/libraries/HMXLib.sol";
+import { OwnableUpgradeable } from "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import { PythStructs } from "pyth-sdk-solidity/IPyth.sol";
+import { ICIXPythAdapter } from "./interfaces/ICIXPythAdapter.sol";
+import { IReadablePyth } from "./interfaces/IReadablePyth.sol";
+
+/// @dev Customized Index Pyth Adapter - Index price will be calculated using geometric mean according to weight config
+contract CIXPythAdapter is OwnableUpgradeable, ICIXPythAdapter {
```

Comment:
I think this design might not be very gas effective as every read needs to calculate the index price.

I'm leaning towards make this as a EcoPythCalldataBuilder and feed their instead.

Reply:
Will turn this into `CIXPriceLens`, and plug it beside `OnChainPriceLens` at `EcoPythCalldataBuilder3`.